packages/ens-controller/src/EnsController.ts

```diff
@@ -2,11 +2,36 @@ import {
   BaseControllerV2,
   RestrictedControllerMessenger,
 } from '@metamask/base-controller';
+import {
+  Web3Provider,
+  ExternalProvider,
+  JsonRpcFetchFunc,
+} from '@ethersproject/providers';
 import {
   normalizeEnsName,
   isValidHexAddress,
   toChecksumHexAddress,
+  NETWORK_ID_TO_ETHERS_NETWORK_NAME_MAP,
+  convertHexToDecimal,
+  hasProperty,
 } from '@metamask/controller-utils';
+import { toASCII } from 'punycode/';
+import ensNetworkMap from 'ethereum-ens-network-map';
+
+/**
+ * Checks whether the given string is a known network ID.
+ *
+ * @param networkId - Network id.
+ * @returns Boolean indicating if the network ID is recognized.
+ */
+function isKnownNetworkId(
+  networkId: string | null,
+): networkId is keyof typeof NETWORK_ID_TO_ETHERS_NETWORK_NAME_MAP {
```

Comment:
Interesting; this error doesn't show up in VSCode, only upon building.

It seems to be caused by the keys being interpreted as numbers for some reason, even though they're defined as strings. Not sure why.

I was able to resolve it by converting the network ID constant to an enum, then declaring `NETWORK_ID_TO_ETHERS_NETWORK_NAME_MAP` as type `Record<NetworkId, NetworkType>`